store/multi_writer_app_store.go

```diff
@@ -225,7 +225,8 @@ func (s *MultiWriterAppStore) pruneOldEVMKeys() error {
 	}
 
 	maxKeysToPrune := cfg.GetAppStore().GetNumEvmKeysToPrune()
-	if maxKeysToPrune > 0 {
+	pruneInterval := cfg.GetAppStore().GetPruneEvmKeysInterval() // Prune old EVM Keys every N blocks
+	if maxKeysToPrune > 0 && (pruneInterval == 0 || s.Version()%int64(pruneInterval) == 0) {
```

Comment:
I think this would be a bit clearer?
```golang
// If pruneInterval is set then only prune old EVM Keys every N blocks
if (pruneInterval != 0) && (s.Version()%int64(pruneInterval)) != 0) {
  maxKeysToPrune = 0
}
if maxKeysToPrune {
```